[moguries] iP

moguries opened the pull request:
# Fatnom Chatbot 
> "Make your future self proud." - Fatnom

≽/ᐠ - ˕ -マ≼

With Fatnom, you can keep track of your daily tasks. It is:

- text-based
- simple
- **_super_** fast to use!

Install it right now for **FREE**!
1. download it [here](https://github.com/moguries/ip/releases)
2. move it to your desired folder
3. open a command window in that folder
4. run the command `java -jar Duke.jar`

Features:
- [x] Manage your task list

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

What are you waiting for? Download it today!

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

moguries committed:
Level-0

moguries committed:
Level-1

moguries committed:
Level-2

moguries committed:
Level-3

moguries committed:
Level-4

moguries committed:
A-TextUiTesting

moguries committed:
Level-5

moguries committed:
Level-6

moguries committed:
Add Storage.java

moguries committed:
Merge Level-7

Merge branch 'branch-Level-7'

moguries committed:
Store dates and times as LocalDateTime objects

moguries committed:
Merge branch 'branch-Level-8'

Merge branch 'branch-Level-8'

Fidget-Spinner commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,48 @@
+     * @param time The LocalDateTime object to be formatted.
+     * @return The formatted deadline, as a String.
+     */
+    public String deadlineFormatter(LocalDateTime time) {
```
Perhaps this could be named
```suggestion
    public String formatDeadline(LocalDateTime time) {
```
instead (to follow the verb way) ?

moguries replied:
Thanks for pointing it out! I've changed it :D

Fidget-Spinner commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,44 @@
+     * @param time The LocalDateTime object to be formatted.
+     * @return The formatted deadline, as a String.
+     */
+    public String deadlineFormatter(LocalDateTime time) {
```
Again, maybe name it `formatDeadLine`?

Fidget-Spinner reviewed (commented):
Hello! This is my second review for CS2103T ip. Overall, the naming looks good. I just have a few minor comments.

wapisai commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,300 @@
+     *
+     * @return The line.
+     */
+    public static String createLine() {
+        String line = "";
+        for (int i = 0; i < LINE_LENGTH; i++) {
+            line += "_";
+        }
+        return line;
+    }
```
Would it be better to just have a `private static final String = "_____";` instead of this method?

wapisai commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,48 @@
+     * @param time The LocalDateTime object to be formatted.
+     * @return The formatted deadline, as a String.
+     */
+    public String deadlineFormatter(LocalDateTime time) {
```
Agree with @Fidget-Spinner

wapisai commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,300 @@
+                } else if (command.equalsIgnoreCase("unmark")) {
+                    int taskNum = Integer.parseInt(inputTokens[1]);
+                    if (taskNum == 0) {
+                        String exceptionMessage = Duke.createLine() + "\n"
+                                + "task 0? how can i unmark a task that doesn't exist?!\n"
```
Good that you have prepared for so many errors/situations!

wapisai commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,136 @@
+ * @version CS2103T AY24/25 Semester 1, G07
+ */
+public class Storage {
+    private static final String PATH_DIRECTORY = "./data/";
+    private static final String PATH_FILE = "./data/duke.txt";
+    private static final Pattern PATTERN_TODO = Pattern.compile("([A-Z]) \\| (\\d) \\| (.*?)");
+    private static final Pattern PATTERN_DEADLINE = Pattern.compile("([A-Z]) \\| (\\d) \\| (.*?) \\| (.*?)");
+    private static final Pattern PATTERN_EVENT = Pattern.compile("([A-Z]) \\| (\\d) \\| (.*?) \\| (.*?) \\| (.*?)");
+    private static DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM dd yyyy h:mma");
```
Good naming!

wapisai commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,136 @@
+    }
+
+    /**
+     * Parses the inputted line for the task data.
```
Perhaps "input" rather than "inputted"? But no big deal

moguries replied:
Thanks for the review :D
haha no not my dog just a made up character.. I imagine it to be a fat cat!

wapisai reviewed (commented):
Overall good coding standard and only one small lapse!!
Also is fatnoms your dog or something because that is cute

moguries committed:
Add more OOP

moguries committed:
Add more OOP

OOP paradigm can provide better abstraction and organisation for Duke.

Extract Ui, Storage, Parser and TaskList classes.

Merge branch 'branch-A-MoreOOP'

moguries committed:
Add packages

moguries committed:
Organise classes into packages

Every class should be part of a package.

Add packages exception, parser, storage, task and ui under main package chatbot.

Merge branch 'branch-A-Packages'

moguries committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Merge is necessary to add gradle support.

moguries committed:
Add JUnit tests

moguries committed:
Add JUnit tests

We must test the behaviour of the code with JUnit.

Add ParserTest and TaskListTest classes.

Merge branch 'branch-A-JUnit'

moguries committed:
Package app as JAR file

moguries committed:
Package the app as a JAR file

The app should be packaged and distributed for use.

Merge branch 'branch-A-Jar'

moguries committed:
Add JavaDoc

moguries committed:
Merge branch 'branch-A-JavaDoc'

moguries committed:
Follow the coding standard

moguries committed:
Merge branch 'branch-A-CodingStandard'

moguries committed:
Implement /find/ command

moguries committed:
Merge branch 'branch-Level-9'

moguries committed:
Implement GUI

moguries committed:
Merge branch 'branch-Level-10'

moguries committed:
Use assertions

There are no assertions being used.

We should document important assumptions that should hold at various points in the code, to use as debugging aids.

Let's add assertions for methods in the TaskList, Parser, Storage and Task classes.

These classes contain methods where we can do input validation with assertions.

moguries committed:
Improve code quality

There are small lapses in the coding standard and quality.

Improving the code quality can enhance readability of the code.

Make adjustments to the structure of the code and naming of variables, as well as add more JavaDoc comments.

This enhances the code quality and ensures that we comply with the Java coding standard.

moguries committed:
Merge pull request #3 from moguries/branch-A-CodeQuality

Improve code quality

moguries committed:
Add Help extension

There is no help page.

Having a help page provides users with a consolidated view of all available commands and command formats. It is especially beneficial for first-time users.

Add a Help Page that is accessible with the command "help".

Users can easily access the help page with a simple command.

moguries committed:
Merge branch 'branch-BCD-Extension'

moguries committed:
Add personality

moguries committed:
Merge branch 'branch-A-Personality'

moguries committed:
Make better GUI

moguries committed:
Merge branch 'branch-A-BetterGui'

moguries committed:
Add user guide

moguries committed:
Merge branch 'branch-A-UserGuide'

moguries committed:
Release latest Jar file

moguries committed:
Merge branch 'branch-A-Release'

moguries committed:
Update README.md
